api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -456,6 +456,16 @@ type HostedClusterSpec struct {
 	// +required
 	Platform PlatformSpec `json:"platform"`
 
+	// KubeAPIServerDNSName specifies a desired DNS name to resolve to the KAS.
+	// This can only be set if secretRef is also set. When set, the controller will automatically generate a secret with kubeconfig and expose it in the hostedCluster Status.
```

Comment:
This comment doesn't seem accurate anymore?

Can we please also document in this API field how the kubeconfig generation it triggers will consume configuration.apiserver.ca/certs?